### DISCUSSION TITLE
Scrollbar issue with custom_scrollbars.uc.js on Firefox 127+

### DISCUSSION THREAD

**Discussion by NoNameNeeded5** [Aris-t2/CustomJSforFx#113] [Q&A]
I think there is a little problem with the custom scrollbar script on Firefox 127+

I've modified the script so that the scrollbar looks like a classic scrollbar (when you use the classic theme in Windows).
The script itsself still works in Firefox 127+ but the scrollbar is displayed without its 3D edges and it
s also missing the lower scrollbar button (the thing with the arrow on it).
However, the scrollbar still looks perfectly fine in other parts of the browser (like the sidebar panel or the libary window), the problem only occurs in the browser window.
So modifying the code would probably ruin the scrollbars in the other parts of the browser.

Here's a screenshot of the browser and the scrollbars.

![scrollbar issue](https://github.com/user-attachments/assets/62e15dab-23ce-4814-ad12-41d67a1c85ca)

And here's my modified version of the custom scrollbar script

[scrollbar113.uc.js.txt](https://github.com/user-attachments/files/16322621/scrollbar113.uc.js.txt)

Maybe there is a about:config setting that will screw up the look of the scroll bar or I don't know...
I couldn't find anything.

**Comment by Aris-t2** [comment]
Try to setup everything again using the latest version of the script https://raw.githubusercontent.com/Aris-t2/CustomJSforFx/master/scripts/custom_scrollbars.uc.js (2.0.5, Aug 17, 2023)

Your file is based on version 2.0.2 (Apr 2, 2023)

**Comment by NoNameNeeded5** [comment]
I did that now but unfortunately it didn't change anything about the problem.

Here's the new file:
[scrollbarnew.uc.zip](https://github.com/user-attachments/files/16382851/scrollbarnew.uc.zip)

**Comment by NoNameNeeded5** [comment]
It would be great if this could be fixed.

It has to do with something that only affects the browser window.

**Comment by Aris-t2** [comment]
I think you have to use a different approach here. 
Test this:
[custom_scrollbars.uc.js.zip](https://github.com/user-attachments/files/16511782/custom_scrollbars.uc.js.zip)

![image](https://github.com/user-attachments/assets/bf0ed61f-d969-44e1-aadb-5494274b88ca)

**Comment by NoNameNeeded5** [comment]
I think I understand what you did there.
You removed the 3d highlight and just left the shadow (as dark lines on the bottom and the right hand side of the thumb) and even though it looks a bit more like a 3D bar, the highlight is of course now missing...

I really wonder what they did there.
Is it that "ThreeDHighlight" no longer works in the browser part of Firefox?

I don't know how familiar you are with the development of the browser but did you read anything like that?
Maybe there would be a way to set it back to its former state?


And for some strange reason, the lower arrow button is still missing in my case...
I don't know why that is.
But what I did find out is that it WILL reappear if you set the browser window to an extremely high zoom setting.

**Comment by NoNameNeeded5** [comment]
Little update:
So the main problem is that colors like ThreeDHighlight or ThreeDShadow are now ignored in the browser window, even though they're still valid options in every other Firefox window.
Very clever and useful.
Not.

But at least I was now able to fix the problem (I used the color white instead of ThreeDHighlight), so thx for your help but the problem that remains is that the lower scrollbar button is missing.
It's only visible when I zoom the page a lot, which is strange.

**Comment by NoNameNeeded5** [comment]
Ok, now I also know why the lower scrollbar button isn't visible.
It's hidden by the status bar...
Great.

Do you have any idea how I could modify the status bar css file to show the scrollbar button?

[statusbar107.zip](https://github.com/user-attachments/files/16516866/statusbar107.zip)

**Reply by Aris-t2** [reply]
How does your statusbar hide it? Using the bottom toolbar myself here for this test and buttons look fine:
![image](https://github.com/user-attachments/assets/f67a835d-efbe-4748-b441-8adc42e09c7a)
![image](https://github.com/user-attachments/assets/1ee5b912-975f-4040-a174-7d3e1006f230)

**Comment by NoNameNeeded5** [comment]
Look at the screenshot in my first post.

There is a status bar and the scrollbar button is hidden by the status bar.

I think there is also some code in the status bar css file to prevent that from happening.
But with the changes that occurred with Firefox 127+ it no longer works.

/* Adds bottom margin to window for status panel will not interfere with scrollbar. 
Can be omitted when status panel is displayed dynamically */
.browserContainer, #main-window {  
	border-bottom: 20px solid var(--lwt-accent-color) !important;    
}

This code no longer seems to work.

**Comment by NoNameNeeded5** [comment]
Well, the above code actually does work but it doesn't do what it's supposed to.
(the only thing why it didn't work was the "color")

I really wish Mozilla would stop making changes that no one actually needs but will affect any customization drastically.
Makes no sense.

**Comment by NoNameNeeded5** [comment]
Any chance to fix this?

**Reply by Aris-t2** [reply]
Sorry, I do not have  another workaound.
